Uncomment and fix useGridEventMouseDown test expect statement where o…

MerlinTheWhiz opened the pull request:
Changes:
- Re-enabled expect(onDrag).not.toHaveBeenCalled() assertion in the "should not trigger drag when mouse moves" test case
- Ensures proper validation of drag prevention when form is open

Closes #316

MerlinTheWhiz commented:
@tyler-dane Hey 👋 trust you're doing fine. Would you mind reviewing the PR and giving me feedback? Sorry for not notifying you by tagging you like this earlier, I thought the PR link on the issue would be enough

that-one-arab commented:
@MerlinTheWhiz Hello! I'll help you review the PR.
Can you explain to me why we needed to update some dependancies and introduce a new dependency? (`ts-jest`)

MerlinTheWhiz commented:
@that-one-arab  Hi!

I updated the dependencies and introduced ts-jest to fix test execution issues related to TypeScript.
While working on the test case should not trigger drag when mouse moves, I encountered a SyntaxError: Missing semicolon, which was tied to TypeScript type assertions like as Schema_GridEvent. The existing setup wasn’t fully equipped to handle those assertions.
Adding ts-jest resolved this by properly compiling TypeScript test files and handling type annotations. Some existing dependencies were also updated to ensure compatibility with ts-jest and resolve peer dependency warnings.
These changes were necessary to get the tests running correctly after uncommenting the expect(onDrag).not.toHaveBeenCalled() assertion.

Let me know if you'd like any changes or prefer an alternate approach!

that-one-arab commented:
@MerlinTheWhiz How did you test this? Because from my end the uncommented test still fails

```
  useGridEventMouseDown
    ✓ should trigger click when mouse releases before hold delay (8 ms)
    ✓ should trigger drag when mouse stays down after hold delay (2 ms)
    ✓ should trigger drag when mouse moves before hold delay (2 ms)
    When form is open
      ✓ should trigger drag when mouse is held down after delay (1 ms)
      ✓ should not trigger drag when mouse is released before delay (2 ms)
      ✕ should not trigger drag when mouse moves (2 ms)

  ● useGridEventMouseDown › When form is open › should not trigger drag when mouse moves

    expect(jest.fn()).not.toHaveBeenCalled()

    Expected number of calls: 0
    Received number of calls: 2
```

that-one-arab commented:
@MerlinTheWhiz Closing due to inactivity. Please feel free to re-open after resuming work on it
